src/distilabel/steps/base.py

```diff
@@ -43,6 +46,37 @@
 extra metadata that allows `distilabel` to perform validations over the `process` step
 method defined in each `Step`"""
 
+# Pattern to convert PascalCase to snake_case
+PATTERN_PASCAL_NAME = re.compile(r"(?<!^)(?=[A-Z])")
+
+
+def _infer_step_name(step_cls_name: str, pipeline: Optional["Pipeline"]) -> str:
```

Comment:
```suggestion
def _infer_step_name(step_cls_name: str, pipeline: Optional["Pipeline"] = None) -> str:
```

Reply:
Here the type could have been `Pipeline | None`, the thing is that if this is run in a single step, the pipeline argument could be `None`. But the change makes sense anyway